lib/rspec/expectations/failure_aggregator.rb

```diff
@@ -48,6 +48,8 @@ def call(failure, options)
         @seen_source_ids[source_id] = true
         assign_backtrace(failure) unless failure.backtrace
         failures << failure
+
+        AggregatedFailure.build(failure)
```

Comment:
My point about a magic object was to avoid an instantiation here, its simply to differentiate that its not a random nil being returned from the object, but a failed method call